resources/qml/ApplicationSwitcher/ApplicationSwitcherPopup.qml

```diff
@@ -99,7 +99,24 @@ Popup
                 iconSource: modelData.thumbnail
                 tooltipText: modelData.description
                 isExternalLink: true
-                visible: modelData.DFAccessRequired ? Cura.API.account.isLoggedIn & Cura.API.account.additionalRights["df_access"] : true
+                visible:
+                {
+                    try
+                    {
+                        modelData.permissionsRequired.forEach(function(permission)
+                        {
+                            if(!Cura.API.account.isLoggedIn || !Cura.API.account.permissions.includes(permission)) //This required permission is not in the account.
+                            {
```

Comment:
```suggestion
                            if (!Cura.API.account.isLoggedIn || !Cura.API.account.permissions.includes(permission)) // This required permission is not in the account.
```

Reply:
This style is not in our code style guidelines (not specified) and I don't really agree with it, so I'll leave it as is.